tpe/sampler.go

```diff
@@ -404,12 +404,11 @@ func (s *Sampler) sampleInt(distribution goptuna.IntUniformDistribution, below,
 
 func (s *Sampler) sampleDiscreteUniform(distribution goptuna.DiscreteUniformDistribution, below, above []float64) float64 {
 	q := distribution.Q
-	r := distribution.High - distribution.Low
-	// [low, high] is shifted to [0, r] to align sampled values at regular intervals.
-	low := 0 - 0.5*q
-	high := r + 0.5*q
-	best := s.sampleNumerical(low, high, below, above, q, false) + distribution.Low
-	return math.Min(math.Max(best, distribution.Low), distribution.High)
+	low := distribution.Low - 0.5*q
+	high := distribution.High + 0.5*q
+	best := s.sampleNumerical(low, high, below, above, q, false)
+	clipped := math.Min(math.Max(best, distribution.Low), distribution.High)
+	return clipped
```

Comment:
Although the performance was significantly improved. I think these code changes are dominant.